Lib/test/test_keywordonlyarg.py

```diff
@@ -63,7 +63,8 @@ def f(a, b=None, *, c=None):
             pass
         with self.assertRaises(TypeError) as exc:
             f(1, 2, 3)
-        expected = "f() takes from 1 to 2 positional arguments but 3 were given"
+        expected = (f"{f.__qualname__}() takes from 1 to 2 "
+                    "positional arguments but 3 were given")
```

Comment:
It would be good to provide a failing test case for each code path you've changed.